src/services/shopify-client.ts

```diff
@@ -0,0 +1,23 @@
+import { EmptyResult } from '../doppler-types';
+export enum SubscriberListState {
+  ready,
+  synchronizingContacts,
+}
+
+export interface SubscriberList {
+  name: string;
+  id: number;
+  amountSubscribers: number;
+  state: SubscriberListState;
+}
+
+export interface ConnectedShops {
+  shopName: string;
+  synchronization_date: Date | null;
+  list: SubscriberList;
+}
+
+export type ShopifyErrorResult = { cannotConnectToAPI: true };
+export interface ShopifyClient {
+  getShopifyData(): Promise<ConnectedShops[] | EmptyResult<ShopifyErrorResult>>;
```

Comment:
Creo que el tipo de resultado debería ser `Result<ConnectedShops[], ShopifyErrorResult>`

```suggestion
  getShopifyData(): Promise<Result<ConnectedShops[], ShopifyErrorResult>>;
```

Reply:
This should behave in the same way right?

Reply:
So so, it will enforce you to wrap the result and include the `success: true` field, like here:

https://github.com/FromDoppler/doppler-webapp/blob/7b5399b8ec18d8fbbf94032ab75e2f3bc7428af9/src/services/doppler-legacy-client.ts#L321-L328

Reply:
I've tried but for some reason it gives error. 
![image](https://user-images.githubusercontent.com/2439363/59516292-6eb3e500-8e97-11e9-9c88-9d87e2eae9e2.png)

Reply:
It is not empty any more hehe

Reply:
please, let me know if it is not clear, we can talk about it.

Reply:
Yeah I've noticed it. I'm using `Result` now locally.